### DISCUSSION TITLE
Azure Workload Identity Support

### DISCUSSION THREAD

**Discussion by tkennes** [cloud-carbon-footprint/cloud-carbon-footprint#1148] [Ideas]
Hi,

Potenial new contributor here. 😁 

Perhaps it would be a good idea to start implementing Azure Workload Identity support, which is in line with future development of authentication between k8s resources and cloud resources, not only for Azure. This will also make future implementations much more secure since you do not have to worry about passing around client secrets for a user/spn that potentially has a lot of permissions within the cloud. Therefore, a huge win for security, and thus for future adoption. 😄 

I believe we can achieve it by extending https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/blob/trunk/packages/azure/src/application/AzureCredentialsProvider.ts in this fashion https://github.com/Azure/azure-workload-identity/blob/main/examples/azure-identity/node/index.js, shall I pick it up?

Tom

**Comment by ericksod** [comment]
Hey @tkennes,
Welcome to the community!
We agree that sounds like a good idea and if you're open to working on it, please do! We just created an issue linked to this discussion #1152 . If you could comment on it, then we can assign it to you and get it on the board. You can comment on it as you work or reach back out if you need assistance.
Thanks!
